lib/brains/tests/test_forms.py

```diff
@@ -51,23 +54,77 @@ def test_parse_ok(self):
         )
 
 
-class TestSubscription(BraintreeTest):
+class TestSubscription(BraintreeTest, ProductsTest):
+    gateways = {'sub': SubscriptionGateway}
 
     def setUp(self):
-        self.form = SubscriptionForm()
-
-    def test_default_name(self):
-        eq_(self.form.get_name('not-brick'), 'Product')
+        super(TestSubscription, self).setUp()
+        self.paymethod, seller_product = create_method_all()
+
+    def submit(self, expect_errors=False, data=None):
+        if not data:
+            data = {}
+        submission = {
+            'paymethod': self.paymethod.get_uri(),
+            'plan_id': 'moz-brick',
+        }
+        submission.update(data)
+        form = SubscriptionForm(submission)
+        if not expect_errors:
+            assert form.is_valid(), form.errors.as_text()
+        return form
 
     def test_seller_name(self):
-        eq_(self.form.get_name('moz-brick'), 'Product')
+        eq_(SubscriptionForm().get_name('moz-brick'), 'Recurring')
+
+    def test_plan_is_not_a_seller_product(self):
+        form = self.submit(data={'plan': 'no-seller-product'},
+                           expect_errors=True)
+
+        assert 'plan' in form.errors, form.errors.as_text()
+
+    def test_plan_is_not_a_configured_product(self):
+        seller, prod = create_seller(seller_product_data={
+            'public_id': 'plan-id-not-in-config',
+        })
+        form = self.submit(expect_errors=True,
+                           data={'plan': prod.public_id})
+        assert 'plan' in form.errors, form.errors.as_text()
+
+    def test_plan_amount_is_not_customizable(self):
+        form = self.submit(expect_errors=True, data={
+            # Link to a non-donation product.
+            'plan': 'moz-brick',
+            # You cannot change the price on a product with a fixed amount.
+            'amount': '1.99',
+        })
+        assert 'amount' in form.errors, form.errors.as_text()
```

Comment:
oh yeah, good call. I don't know if I can get at the code but I can pattern match part of the message